### PR TITLE
[Bug fix] VMware: Fix for SSVM recreation on deployasis systemVM templates

### PR DESCRIPTION
**Pull request by nvazquez**
## Description
Fix for VMware SSVM recreation using deploy-as-is system VM templates

Fixes:
- Register a new system VM template
- Recreate SSVM -> ERROR

## Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
VMware environment

- Register a new system VM template
- Recreate SSVM

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test centos7 vmware67-u3

**Comment by nvazquez** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Review by Pearl1594 — APPROVED** [review]
Verified fix. LGTM

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java [hunk 1]
```diff
@@ -824,7 +824,7 @@ private DiskProfile allocateTemplatedVolume(Type type, String name, DiskOffering
         int volumesNumber = 1;
         List<DatadiskTO> templateAsIsDisks = null;
         String configurationId = null;
-        if (template.isDeployAsIs()) {
+        if (template.isDeployAsIs() && vm.getType() != VirtualMachine.Type.SecondaryStorageVm) {
```
Comment (harikrishna-patnala) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:827]: do we have to also keep a check for other system VMs ? or pass this condition only for user VMs ? I see a problem when CPVM is tried to recreate when SSVM is not there, we might get into same issue.

Comment (rhtyd) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:827]: Good point; @nvazquez can you check for CPVM and other systemvm types as well?

**Review by rhtyd — APPROVED** [review]
Approach LGTM, once the issue wrt other systemvm types is resolved we can test and merge. (I did n't manually test it)

**Comment by nvazquez** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java (hunk 1 above)
Comment (nvazquez) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:832]: I've pushed a fix to check for running SSVMs in the zone as well

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test centos7 vmware-67u3

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java (hunk 1 above)
Comment (harikrishna-patnala) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:832]: Thanks for fixing that case @nvazquez

**Review by harikrishna-patnala — APPROVED** [review]

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by rhtyd** [comment]
Let's merge once this passes, pl check if the last test env failed due to a regression @nvazquez thnx

**Comment by nvazquez** [comment]
@rhtyd tests running now, previous env failed to start the marvin VM

**Pull request merged**